Core.UnitTests/ReadInput/InputReaderTests.cs

```diff
@@ -28,6 +28,67 @@ namespace Remotion.ReleaseProcessAutomation.UnitTests.ReadInput;
 [TestFixture]
 internal class InputReaderTests
 {
+  [Test]
+  [TestCase("y", true)]
+  [TestCase("Y", true)]
+  [TestCase("n", false)]
+  [TestCase("N", false)]
+  public void ReadConfirmation_WithInput_ReturnsOutput (string inputText, bool expectedOutput)
+  {
+    var testConsole = new TestConsole();
+    testConsole.Input.PushTextWithEnter(inputText);
+    var inputReader = new InputReader(testConsole);
+
+    var result = inputReader.ReadConfirmation();
+
+    Assert.That(result, Is.EqualTo(expectedOutput));
+  }
+
+  [Test]
+  public void ReadConfirmation_WithFirstIllegalInput_WithSecondOKInput_AsksUserAgainAfterFirstInput ()
+  {
+    var testConsole = new TestConsole();
+    testConsole.Input.PushTextWithEnter("notACorrectInput");
+    testConsole.Input.PushTextWithEnter("y");
+    var inputReader = new InputReader(testConsole);
+
+    var result = inputReader.ReadConfirmation();
+
+    var lines = testConsole.Output.Split("\n");
+    Assert.That(lines[0], Is.EqualTo("Confirm? [y/n] (y): notACorrectInput"));
+    Assert.That(lines[1], Is.EqualTo("The input 'notACorrectInput' is not a valid option."));
+    Assert.That(lines[2], Is.EqualTo("Confirm? [y/n] (y): y"));
+    Assert.That(result, Is.EqualTo(true));
+  }
+
+  [Test]
+  [TestCase(true)]
+  [TestCase(false)]
+  public void ReadConfirmation_WithDifferingDefaultValues_ChangesYesNoAtEndOfMessageDependingOnDefaultValue (bool defaultValue)
+  {
+    var testConsole = new TestConsole();
+    testConsole.Input.PushTextWithEnter("y");
+    var inputReader = new InputReader(testConsole);
+
+    var result = inputReader.ReadConfirmation(defaultValue);
+
+    Assert.That(testConsole.Output, Does.Contain($"Confirm? [y/n] ({(defaultValue ? "y" : "n")}):"));
```

Comment:
Test refactoring:
it's best practice to have as little conditinal logic in the test as possible. In particular you want to avoid duplicating the logic you are testing. Here, if you mismatch the true/y and false/n combinations when evaluating the default value, the probability is high that you also mismatched in the production code, which uses the same condition. Therefore, its best to just make the entire string a testcase-parameter. If the entire string is too long (which isn't really the case here), string concat is fine. 

Unless there's a good reason against it, please push the string into the testcase.

Reply:
Ok, thanks for the advice :)